### ISSUE TITLE
PropertyGrid accessible objects are leaking in memory

### ISSUE THREAD

**Issue by ArtemTatarinov** [dotnet/winforms#6943]
### .NET version

7.0

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

`PropertyGrid` accessible objects are left in memory after the `PropertyGrid` control instance was inspected with accessibility tool (like `Accessibility Insights` or `Inspect`) and the form with `PropertyGrid` is closed, garbage is collected, but accessibility tool is still opened.

Report from WinDbg:
![image](https://user-images.githubusercontent.com/87859299/161093704-221951bf-737d-4f52-a4d6-afb03a9c95a1.png)

**Expected behavior:**
Amount of the leaked objects should be minimized to the uncleanable static objects only.

### Steps to reproduce

1. Create a parent form with two buttons: one will be opening a child form with the `PropertyGrid` (`SelectedObject` set to `this`) control and the other will collect the garbage.
2. Press the second button to open the child form with `PropertyGrid`.
3. Start the `Accessibility Insights` tools.
4. Run automated checks for the `PropertyGrid` control using AI.
5. Close the child form.
6. Clear memory with the first button of the parent form.
7. DON'T close the AI tool.
8. Look for `PropertyGrid` accessible objects left in memory (using `WinDbg`, or `dotMemory`, or another suitable tool).

**Comment by John-Qiao** [comment]
Verified with .NET 8.0.100-preview.2.23101.9 build from main branch, this issue is fixed as below:
![6943-testresult](https://user-images.githubusercontent.com/45864985/216230030-92bf84e0-8004-4cf1-a9b2-05f72a665181.png)

**Comment by Cassie-Li01** [comment]
Verified on 8.0 TP version .NET 8.0.100-preview.1.23105.3, issue is fixed. Screenshot same as above.

**Comment by Cassie-Li01** [comment]
Verified with .NET 8 Preview 1 TP build: .NET 8.0.100-preview.1.23112.5, issue is fixed. Screenshot as above
